src/backend/joanie/core/models/products.py

```diff
@@ -944,9 +944,26 @@ def submit_for_signature(self, user: User):
         to submit.
         """
         if not self.product.contract_definition_id:
+            logger.error(
+                "No contract definition attached to the contract's product.",
```

Comment:
This string should be the same used in the exception. You can maybe add it in a variable and then use it at both places ?